src/models/ThemeProvider.js

```diff
@@ -117,7 +114,7 @@ class ThemeProvider extends Component<ThemeProviderProps, void> {
       }
       return mergedTheme
     }
-    if (!isPlainObject(theme)) {
+    if (!isObject(theme)) {
```

Comment:
Looks good. Could you add a test that simulates what webpack does to the best of your ability so we know it'll work?